### PR TITLE
feat: gossip client swarm bind

### PR DESCRIPTION
**Pull request by alanshaw**
This PR adds a function that periodically connects to peers passed in the `peer-with` param. This ensures that if we get disconnected from a gossipsub drand relay we'll reconnect again.

### CODE REVIEW

**Comment by Kubuxu** [comment]
`WithDirectPeers` from pubsub creation should be already doing that.

File: cmd/relay-gossip/lp2p/swarm.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+func SwarmBind(ctx context.Context, h host.Host, bindAddrs []ma.Multiaddr, period time.Duration) {
+	go func() {
+		for {
+			t := time.NewTimer(period)
+			select {
+			case <-t.C:
+				for _, ma := range bindAddrs {
+					ai, err := peer.AddrInfoFromP2pAddr(ma)
+					if err != nil {
+						fmt.Println("invalid p2p address", err)
```
Comment (petar) [inline cmd/relay-gossip/lp2p/swarm.go:23]: log.Println

**Comment by alanshaw** [comment]
I thought `WithDirectPeers` required reciprocal configuration on the relay?

**Comment by alanshaw** [comment]
At least, when I stopped the relay yesterday and started it back up again the client did not re-establish a connection.

**Comment by Kubuxu** [comment]
Without reciprocal configuration it does not guarantee that relay will let you in into a mesh, but client should reconnect after some time (looking at default gossip config it is 300min....

**Comment by Kubuxu** [comment]
Sorry, it isn't 300min but 300s.

**Comment by alanshaw** [comment]
Ah I definitely didn’t wait that long...!

**Pull request closed without merging**